### PR TITLE
Add support for web config file

### PR DESCRIPTION
**Pull request by SuperQ**
[feature] Add support for the web config file to allow https/auth
configuration.

Signed-off-by: Ben Kochie <superq@gmail.com>

### CODE REVIEW

**Comment by SuperQ** [comment]
Finally figured out the problem. The alternative test is injecting the prometheus version in a funny way, without setting the variable needed to key the `if`.

File: molecule/alternative/playbook.yml [hunk 1]
```diff
@@ -5,6 +5,8 @@
   roles:
     - ansible-prometheus
   vars:
+    # This version should match prepare.yml
+    prometheus_version: 2.14.0
```
Comment (roidelapluie) [inline molecule/alternative/playbook.yml:9]:
```suggestion
    prometheus_version: 2.24.0
```

Comment (paulfantom) [inline molecule/alternative/playbook.yml:9]: The idea was that alternative tests are downloading prometheus binary in `prepare.yml` playbook and then use a binary as a local one. For such mode of operation `prometheus_version` shouldn't matter as it is set to download the correct tarball in `download prometheus binary to local folder` task.

However, it seems we need to adjust this mode of operation as we are also using `prometheus_version` in systemd service template.

File: molecule/alternative/prepare.yml [hunk 2]
```diff
@@ -3,14 +3,15 @@
   hosts: localhost
   gather_facts: false
   vars:
-    # Version seeds to be specified here as molecule doesn't have access to ansible_version at this stage
-    version: 2.14.0
+    # Version seeds to be specified here as molecule doesn't have access to prometheus_version at this stage.
+    # It needs to match playblook.yml
```
Comment (paulfantom) [inline molecule/alternative/prepare.yml:7]:
```suggestion
    # It needs to match playbook.yml
```

File: molecule/alternative/prepare.yml [hunk 3]
```diff
@@ -3,14 +3,15 @@
...
+    prometheus_version: 2.14.0
   tasks:
     - name: download prometheus binary to local folder
       become: false
       get_url:
-        url: "https://github.com/prometheus/prometheus/releases/download/v{{ version }}/prometheus-{{ version }}.linux-amd64.tar.gz"
-        dest: "/tmp/prometheus-{{ version }}.linux-amd64.tar.gz"
+        url: "https://github.com/prometheus/prometheus/releases/download/v{{ prometheus_version }}/prometheus-{{ prometheus_version }}.linux-amd64.tar.gz"
```
Comment (paulfantom) [inline molecule/alternative/prepare.yml:13]: This is on purpose not set to `prometheus_version` so it won't pick up the version from `defaults`.

File: molecule/alternative/playbook.yml (hunk 1 above)
Comment (SuperQ) [reply molecule/alternative/playbook.yml:9]: Ahh, ok. Now I get it.

I guess a solution would be to keep the alternative test version up to at least the oldest supported version.

Comment (paulfantom) [reply molecule/alternative/playbook.yml]: yes, that sounds good :+1:

**Review by roidelapluie — APPROVED** [review]

**Review by paulfantom — APPROVED** [review]
lgtm :shipit:

**Pull request merged**